abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1807 @@
+# Proxy-Wasm ABI v0.2.1 specification
+
+---
+
+Status: **FINAL**
+
+---
+
+# Callbacks and Functions
+
+### Callbacks exposed by the Wasm module
+
+All callbacks (entry points named `proxy_on_<event>`) are optional,
+and are going to be called only if exposed by the Wasm module.
+
+All callbacks, other than the integration functions, include a unique
+context identifier as the first parameter, which can be used to
+distinguish between different contexts.
+
+
+### Functions exposed by the host environment
+
+All functions exposed by the host return [`proxy_status_t`], which
+indicates status of the call (success, invalid memory access, etc.),
+and return values are written into memory pointed by `return_<value>`
+parameters.
+
+
+## Integration
```

Comment:
maybe "guest exports"? This will make more clear which side is supposed to export these? Or maybe say so as a summary comment below.

Reply:
The initial vNEXT spec had the whole document split into 2 sections (guest and host exports) (e.g. [link](https://github.com/proxy-wasm/spec/tree/442687bb307ea7c701d5744b3f0f3e8308010238/abi-versions/vNEXT#functions-implemented-in-the-wasm-module)).

The improved vNEXT that was under review in #1 had each section split into 2 for guest and host exports (e.g. [link](https://github.com/PiotrSikora/spec/blob/cc53262df056036427476b272fb8f2438aa7975f/abi-versions/vNEXT/README.md#functions-exposed-by-the-host-environment-1)).

I removed those this time for a silly reason that some Markdown tools don't render 4th level headings in the sidebar index, so you couldn't jump right to a specific function, and I assumed that it's pretty easy to tell which is which from the `proxy_on_<event>` function names and descriptions (`Called to/when...`), but I'm happy to re-add those (I'll go with split inside each section, like in the improved vNEXT). Thanks!